frontend/app/src/MetricsManager.ts

```diff
@@ -240,4 +291,54 @@ export class MetricsManager {
     }
     return {}
   }
+
+  // Get context data for events
+  private getContextData(): Record<string, unknown> {
+    return {
+      contextPageUrl: window.location.href,
+      contextPageTitle: document.title,
+      contextPagePath: window.location.pathname,
+      contextPageReferrer: document.referrer,
+      contextPageSearch: window.location.search,
+      contextLocale:
+        // @ts-expect-error
+        window.navigator.userLanguage || window.navigator.language,
```

Comment:
nit: I think `navigator.userLanguage` might be a historical relict from Internet Explorer <11. We can probably just use ` window.navigator.language` here (supported by all common browsers) since we are not supporting these < IE11 afaik. This might also allow to remove the `@ts-expect-error`

Reply:
Good catch! I borrowed the logic from Segment's implementation for consistency, though research confirmed its a relic 👍🏼